src/cargo/core/features.rs

```diff
@@ -538,51 +538,73 @@ impl Features {
     }
 }
 
-/// A parsed representation of all unstable flags that Cargo accepts.
-///
-/// Cargo, like `rustc`, accepts a suite of `-Z` flags which are intended for
-/// gating unstable functionality to Cargo. These flags are only available on
-/// the nightly channel of Cargo.
-#[derive(Default, Debug, Deserialize)]
-#[serde(default, rename_all = "kebab-case")]
-pub struct CliUnstable {
+macro_rules! cli_options {
+    (
+        $(#[$struct_meta:meta])*
+        $name: ident,
+        $($visibility: vis $element: ident: $ty: ty = ($help: literal $(, #[$meta:meta])?)),*
```

Comment:
Can the `meta` be moved to before the `visibility` so that it can be written like a natural attribute is done on a struct field?